### ISSUE TITLE
File.open w/ File::RDWR should write \r\n but gets should read as \n after rewind on windows

### ISSUE THREAD

**Issue by enebo** [jruby/jruby#3738]
### Environment

Broken in different ways on JRuby 1.7.24 and 9.0.5.0.

### Expected Behavior

For the following script:

```ruby
file = "temptemp"

File.open(file, "w") {}
File.open(file, File::RDWR) do |f|
  f.puts("writing")  # writes "writing\r\n"
  f.flush            # force to disk
  gets               # do not continue until you can verify it was written right
  f.rewind           # backup
  p f.gets           # read it but it should be "writing\n"
end
File.unlink(file)
```
The comments dictate how it should behave.

### Actual Behavior

On JRuby 1.7.24 it will write "writing\r\n" but it will read "writing\r\n". instead of "writing\n".
On JRuby 9.0.5.0 it wil write "writing\n" instead of "writing\r\n" but it will read "writing\n".

**Comment by enebo** [comment]
Notes since this stuff is scary :) 

we do not use transcoding for CRLF translations but do them through a wrapper independent of transcoding facility.  However, we ported a lot of logic in MRI which assumes this translation will happen in transcoder.  So in doWriteConversion we end up calling needsWriteConversion which returns true because it can see the econv newline stuff is set.  Because this happens we end up setting the open file to be binary which in turn ends up not converting the \r\n to \n on the gets.  Debugger showed that CRLFStreamWrapper is binary on File::RDWR but not when opened up as a plain "r".

**Comment by enebo** [comment]
So in looking at MRI a bit I can see the textmode check does not exist if run on windows.

UNIX:
```C
#define NEED_NEWLINE_DECORATOR_ON_WRITE(fptr) ((fptr)->mode & FMODE_TEXTMODE)
#define NEED_WRITECONV(fptr)
(((fptr)->encs.enc != NULL &&
  (fptr)->encs.enc != rb_ascii8bit_encoding()) ||
    NEED_NEWLINE_DECORATOR_ON_WRITE(fptr) ||
    ((fptr)->encs.ecflags & (ECONV_DECORATOR_MASK|ECONV_STATEFUL_DECORATOR_MASK)))
```

Whereas on Windows:
```C
#define NEED_WRITECONV(fptr)
(((fptr)->encs.enc != NULL &&
  (fptr)->encs.enc != rb_ascii8bit_encoding()) ||
  ((fptr)->encs.ecflags & ((ECONV_DECORATOR_MASK & ~ECONV_CRLF_NEWLINE_DECORATOR)|ECONV_STATEFUL_DECORATOR_MASK)))
```

So I have found a signifcant discrepency.

**Comment by enebo** [comment]
Fixed in commit 4496cffb78f9f97aa568bc1afc574ce8df6cf289.  Not sure why git did not pick this up (perhaps the trailing color after issue number?).
